app/src/main/java/com/nervousfish/nervousfish/modules/database/DatabaseException.java

```diff
@@ -25,4 +29,30 @@ public DatabaseException(final Throwable throwable) {
         super(throwable);
     }
 
+    /**
+     * Deserialize the instance using readObject to ensure invariants and security.
+     *
+     * @param stream The serialized object to be deserialized
+     */
+    private void readObject(final ObjectInputStream stream) throws IOException, ClassNotFoundException {
```

Comment:
I think this is in conflict with your approach to not having to explicitly specify attributes (e.g. explicitly say `boolean myBoolean = false`). But sure, I'll let it go